src/components/action-popover/action-popover.test.tsx

```diff
@@ -0,0 +1,2303 @@
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem>example item 1</ActionPopoverItem>
+        <ActionPopoverItem>example item 2</ActionPopoverItem>
+        <ActionPopoverItem
+          submenu={
+            <ActionPopoverMenu>
+              <ActionPopoverItem>submenu item 1</ActionPopoverItem>
+              <ActionPopoverItem>submenu item 2</ActionPopoverItem>
+            </ActionPopoverMenu>
+          }
+        >
+          example item with submenu
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    const clearTimeoutSpy = jest.spyOn(window, "clearTimeout");
+
+    await user.pointer({
+      target: screen.getByRole("button", { name: "example item with submenu" }),
+    });
+    expect(clearTimeoutSpy).not.toHaveBeenCalled();
+    // need to move the pointer away from the element before moving it back, otherwise mouseEnter won't be triggered a second time
+    await user.pointer({
+      target: screen.getByRole("button", { name: "example item 2" }),
+    });
+    await user.pointer({
+      target: screen.getByRole("button", { name: "example item with submenu" }),
+    });
+
+    expect(clearTimeoutSpy).toHaveBeenCalled();
+    clearTimeoutSpy.mockRestore();
+  });
+
+  it("closes the submenu when the mouse leaves the parent item", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem>example item 1</ActionPopoverItem>
+        <ActionPopoverItem>example item 2</ActionPopoverItem>
+        <ActionPopoverItem
+          submenu={
+            <ActionPopoverMenu>
+              <ActionPopoverItem>submenu item 1</ActionPopoverItem>
+              <ActionPopoverItem>submenu item 2</ActionPopoverItem>
+            </ActionPopoverMenu>
+          }
+        >
+          example item with submenu
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    await user.pointer({
+      target: screen.getByRole("button", { name: "example item with submenu" }),
+    });
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(
+      screen.getByRole("button", { name: "submenu item 1" })
+    ).toBeInTheDocument();
+
+    await user.pointer({
+      target: screen.getByRole("button", { name: "example item 2" }),
+    });
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(
+      screen.queryByRole("button", { name: "submenu item 1" })
+    ).not.toBeInTheDocument();
+  });
+
+  // test needed for coverage of the clearTimeout call - it's not clear if the clearTimeout is actually needed
+  // and if it has any real observable consequences. We should investigate that and remove the call, and this
+  // test, if it is not needed.
```

Comment:
comment: let's make a ticket a for this task so it doesn't get lost

Reply:
now created as FE-6722 (same as above as obviously this isn't worth 2 separate tickets)